python/BioSimSpace/Align/_lomap/lomap_networkgen.py

```diff
@@ -530,11 +535,37 @@ def ecr(mol_i, mol_j):
                         logging.info('MCS molecules: %s - %s' % (self[i].getName(), self[j].getName()))
 
                     if not fingerprint:
-                        # Maximum Common Subgraph (MCS) calculation
-                        MC = mcs.MCS(moli, molj, options=self.options)
-                        ml=MC.all_atom_match_list()
-                        self.set_MCSmap(i,j,ml)
-                        MCS_map[(i,j)]=ml
+                        if ml_pd:
```

Comment:
Do we want to pre-validate the contents of the scoring file in anyway? I know that we check that all ligand pairs are present, but would it also be useful to check that each line contains `str`, `str`, `float` when it is split on commas? Might be useful to do at the start of the `generateNetwork` function where the ligand names are validated, just to catch any surprises. It looks like you would just get  a `KeyError` here if the value was missing. I'm not sure what would happen if the value was a string rather than a float, for example.

Reply:
I think it would be good to validate the contents in some way because it takes a while to trace LOMAP errors back when the input is off.

I'll build that check into `generateNetwork()`; the validation code block is getting quite extensive though. Might need some clean-up at some point.

Reply:
yes, I think the check should be in `_Align.py`, because inputting a str,str,str line in the `scores_file` makes LOMAP fail silently because it just excludes that particular MCS:
`lomap.out`:
```
Skipping MCS molecules (exception): 009_ejm_43.mol2 - 012_ejm_42.mol2	

could not convert string to float:[str]
```

Reply:
see 35138541aa20ea2a9781a95cd868fe24cc4d660d. I allowed users to input ints as well; LOMAP has no problem with scores >1 so in this way it would be fine if a user inputs `scores_file` where scores are any of [0,1,2,..n].

Reply:
I think if you agree with the changes in this one, all should be resolved. Thanks for the feedback!